### PR TITLE
Upgrade Maven plugins, add a Java 9 build on Travis.

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -271,7 +271,7 @@
 
                 <plugin>
                     <artifactId>maven-enforcer-plugin</artifactId>
-                    <version>1.4.1</version>
+                    <version>3.0.0-M1</version>
```
Comment (donraab) [inline pom.xml]: Is this a temporary workaround until there is a GA release?

File: pom.xml [hunk 2]
```diff
@@ -233,7 +233,7 @@
 
                 <plugin>
                     <artifactId>maven-javadoc-plugin</artifactId>
-                    <version>2.10.4</version>
+                    <version>3.0.0-M1</version>
```
Comment (donraab) [inline pom.xml]: Is this a temporary workaround until there is a GA release?

Comment (nikhilnanivadekar) [reply pom.xml]: Yes we waited for about 2 months and we don't know when the plugins will release. I think it's safe to upgrade enforcer and javadoc to milestone version

File: pom.xml (hunk 1 above)
Comment (nikhilnanivadekar) [reply pom.xml]: Yes, explanation same as above

**Pull request merged**